RocketChat/Rocket.Chat.ReactNative
# [NEW] Canned responses

## Proposed changes


Implemented the feature to Rocket.Chat mobile app to support the ability to select/send Canned Responses while the agent is on a room of Omnichannel.

So, here is implemented the feature to get the canned response from input message in room view [Screenshot 1], besides that the agent can see all the canned responses, filter by text and departments in a new view [Screenshot 2], added a new view to see the canned response in detail [Screenshot 3].

## How to test or reproduce


- Be an Omnichannel's agent
- Join an Omnichannel's chat
- In the Room View, write `!` in the message input text
- Should appear a list with canned responses
- Or you can click on title header, go to room actions
- Should appear a button with name: `Canned Responses`
- Click on it, and you should see all the canned responses, that can be filtered by departments and text
- Click on any canned response, should move to detail
- And in both views, if you click the button "USE", you move back to the room view and will appear in message input text the canned response that you selected.

## Screenshots

<img src="https://user-images.githubusercontent.com/47038980/132733417-93fdb834-d62f-4822-9ada-297f51a7a6f5.png" height="450" />

<img src="https://user-images.githubusercontent.com/47038980/132733677-83adf2cf-6927-4899-9969-8f38645792a7.png" height="450" />

<img src="https://user-images.githubusercontent.com/47038980/132733683-824d74a9-cd9c-4bcf-8b3b-dff9c0c924e6.png" height="450" />


https://user-images.githubusercontent.com/47038980/133126944-dc746e04-35e0-494e-9633-a243b208198c.mp4




## Types of changes




## Checklist



## Further comments
